.pre-commit-config.yaml

```diff
@@ -0,0 +1,8 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+- repo: https://github.com/gitleaks/gitleaks
+  rev: v8.18.2
+  hooks:
+    - id: gitleaks-docker
```

Comment:
**question:** Just confirming is this what does the installation?

Reply:
I can't quite tell what is happening under the hood of `pre-commit`, but nothing more is required to be able to run `gitleaks`:

```bash
$ pre-commit 
[WARNING] Unstaged files detected.
[INFO] Stashing unstaged files to /home/vscode/.cache/pre-commit/patch1715609212-4181.
Detect hardcoded secrets.................................................Passed
[INFO] Restored changes from /home/vscode/.cache/pre-commit/patch1715609212-4181.
```